libs/galley-types/src/Galley/Types/Teams.hs

```diff
@@ -544,10 +544,14 @@ instance ToJSON TeamUpdateData where
 
 instance FromJSON TeamUpdateData where
     parseJSON = withObject "team update data" $ \o -> do
-        x <- TeamUpdateData <$> o .:? "name" <*> o .:? "icon" <*> o .:? "icon_key"
-        when (isNothing (_nameUpdate x) && isNothing (_iconUpdate x) && isNothing (_iconKeyUpdate x)) $
-            fail "no update data specified"
-        pure x
+        name     <- o .:? "name"
+        icon     <- o .:? "icon"
+        icon_key <- o .:? "icon_key"
+        when (isNothing name && isNothing icon && isNothing icon_key) $
+            fail "TeamUpdateData: no update data specified"
+        either fail pure $ TeamUpdateData <$> maybe (pure Nothing) (fmap Just . checkedEitherMsg "name")     name
```

Comment:
Right, that is what you get from `checkedEitherMsg`, e.g.:
```
> eitherDecode "{\"name\":\"\"}" :: Either String TeamUpdateData
Left "Error in $: name: outside range [1, 256]"
```

So a request to `PUT /teams/<tid>` will then return
```
{"code":400,"message":"Error in $: name: outside range [1, 256]","label":"invalid-payload"}
```